### DISCUSSION TITLE
Why do I need a token for this?

### DISCUSSION THREAD

**Discussion by Digital-Pig-LLC** [dotkernel/api#335] [General]
In the documentation, there is a page that discusses generating a token for error reporting:
https://docs.dotkernel.org/api-documentation/v5/commands/generate-tokens/

Nowhere in that page does it discuss why I would want to generate a token for error reporting, nor does it explain why I'd want to generate that token and then hardcode it into one of my config files.

What on earth is this for?

**Comment by alexmerlin** [comment]
Being an open-source project, the endpoints in Dotkernel API all well-known to anyone.
Hence, leaving the error reporting endpoint unprotected would allow anyone to bombard it with tons of garbage requests, filling up the server storage in a pretty short time.
That is why it is protected, requiring a token in the header of the request.

**Reply by Digital-Pig-LLC** [reply]
Ah! True true! Thanks for the explanation.
